packages/local_storage_todos_api/lib/src/local_storage_todos_api.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+import 'package:rxdart/subjects.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:todos_api/todos_api.dart';
+
+/// {@template local_storage_todos_api}
+/// A Flutter implementation of the [TodosApi] that uses local storage.
+/// {@endtemplate}
+class LocalStorageTodosApi extends TodosApi {
+  /// {@macro local_storage_todos_api}
+  LocalStorageTodosApi({
+    required SharedPreferences plugin,
+  }) : _plugin = plugin {
+    _init();
+  }
+
+  final SharedPreferences _plugin;
+
+  final _todoStreamController = BehaviorSubject<List<Todo>>.seeded(const []);
+
+  /// The key used for storing the todos locally.
+  ///
+  /// This is only exposed for testing and shouldn't be used by consumers of
+  /// this library.
+  @visibleForTesting
+  static const kTodosCollectionKey = '__todos_collection_key__';
+
+  String? _getValue(String key) => _plugin.getString(key);
+
+  Future<void> _setValue(String key, String value) {
+    return _plugin.setString(key, value);
+  }
+
+  void _init() {
+    final todosJson = _getValue(kTodosCollectionKey);
+    if (todosJson != null) {
+      final todos = List<Map>.from(json.decode(todosJson) as List)
+          .map((jsonMap) => Todo.fromJson(Map<String, dynamic>.from(jsonMap)))
+          .toList();
+      _todoStreamController.add(todos);
+    } else {
+      _todoStreamController.add(const []);
+    }
+  }
+
+  @override
+  Stream<List<Todo>> getTodos() => _todoStreamController.asBroadcastStream();
+
+  @override
+  Future<void> saveTodo(Todo todo) {
+    final todos = [..._todoStreamController.value];
```

Comment:
```suggestion
    final todos = _todoStreamController.value.toList();
```

Alternatively, we could read the value from Shared Prefs again, but I guess the latest value is on the stream as well